python/packages/autogen-core/docs/src/user-guide/core-user-guide/design-patterns/multiple-agents-concurrently.ipynb

```diff
@@ -0,0 +1,313 @@
+     ]
+    }
+   ],
+   "source": [
+    "runtime = SingleThreadedAgentRuntime()\n",
+    "\n",
+    "await Processor.register(runtime, \"agent_1\", lambda: Processor(\"Agent 1\"))\n",
+    "await Processor.register(runtime, \"agent_2\", lambda: Processor(\"Agent 2\"))\n",
+    "\n",
+    "runtime.start()\n",
+    "\n",
+    "await runtime.publish_message(Task(task_id=\"task-1\"), topic_id=DefaultTopicId())\n",
+    "\n",
+    "await runtime.stop_when_idle()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Multiple messages & Multiple Processors\n",
+    "Second, this pattern demonstrates routing different types of messages to specific processors:\n",
+    "- `UrgentProcessor` subscribes to the \"urgent\" topic\n",
+    "- `NormalProcessor` subscribes to the \"normal\" topic\n",
+    "\n",
+    "We make an agent subscribe to a specific topic type using the {py:meth}`~autogen_core.components.type_subscription` decorator."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 122,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@type_subscription(topic_type=\"urgent\")\n",
+    "class UrgentProcessor(RoutedAgent):\n",
+    "    @message_handler\n",
+    "    async def on_task(self, message: Task, ctx: MessageContext) -> None:\n",
+    "        print(f\"Urgent processor starting task {message.task_id}\")\n",
+    "        await asyncio.sleep(1)  # Simulate work\n",
+    "        print(f\"Urgent processor finished task {message.task_id}\")\n",
+    "\n",
+    "\n",
+    "@type_subscription(topic_type=\"normal\")\n",
+    "class NormalProcessor(RoutedAgent):\n",
+    "    @message_handler\n",
+    "    async def on_task(self, message: Task, ctx: MessageContext) -> None:\n",
+    "        print(f\"Normal processor starting task {message.task_id}\")\n",
+    "        await asyncio.sleep(3)  # Simulate work\n",
+    "        print(f\"Normal processor finished task {message.task_id}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After registering the agents, we can publish messages to the \"urgent\" and \"normal\" topics:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 117,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Normal processor starting task normal-1\n",
+      "Urgent processor starting task urgent-1\n",
+      "Urgent processor finished task urgent-1\n",
+      "Normal processor finished task normal-1\n"
+     ]
+    }
+   ],
+   "source": [
+    "runtime = SingleThreadedAgentRuntime()\n",
+    "\n",
+    "await UrgentProcessor.register(runtime, \"urgent_processor\", lambda: UrgentProcessor(\"Urgent Processor\"))\n",
+    "await NormalProcessor.register(runtime, \"normal_processor\", lambda: NormalProcessor(\"Normal Processor\"))\n",
+    "\n",
+    "runtime.start()\n",
+    "\n",
+    "await asyncio.gather(\n",
```

Comment:
No need to use asyncio.gather here because the publish_message doesn't return any result. Calling them one after another is fine.